apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/asyncprofiler/AsyncProfilerTaskExecutionService.java

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.asyncprofiler;
+
+import one.profiler.AsyncProfiler;
+import org.apache.skywalking.apm.agent.core.boot.BootService;
+import org.apache.skywalking.apm.agent.core.boot.DefaultImplementor;
+import org.apache.skywalking.apm.agent.core.boot.DefaultNamedThreadFactory;
+import org.apache.skywalking.apm.agent.core.boot.ServiceManager;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.channels.FileChannel;
+import java.util.Objects;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+@DefaultImplementor
+public class AsyncProfilerTaskExecutionService implements BootService {
+
+    private static final ILog LOGGER = LogManager.getLogger(AsyncProfilerTaskChannelService.class);
+
+    private AsyncProfiler asyncProfilerInstance;
+
+    private static final String SUCCESS_RESULT = "Profiling started\n";
+
+    // profile executor thread pool, only running one thread
+    private ScheduledExecutorService asyncProfilerExecutor;
+
+    // last command create time, use to next query task list
+    private volatile long lastCommandCreateTime = -1;
+
+    // task schedule future
+    private volatile ScheduledFuture<?> scheduledFuture;
+
+    public void processAsyncProfilerTask(AsyncProfilerTask task) {
+        if (task.getCreateTime() <= lastCommandCreateTime) {
+            LOGGER.warn("get repeat task because createTime is less than lastCommandCreateTime");
+            return;
+        }
+        lastCommandCreateTime = task.getCreateTime();
+        LOGGER.info("add async profiler task: {}", task.getTaskId());
+        // add task to list
+        getAsyncProfilerExecutor().execute(() -> {
+            try {
+                if (Objects.nonNull(scheduledFuture) && !scheduledFuture.isDone()) {
+                    LOGGER.info("AsyncProfilerTask already running");
+                    return;
+                }
+
+                String result = task.start(getAsyncProfiler());
+                if (!SUCCESS_RESULT.equals(result)) {
+                    stopWhenError(task, result);
+                    return;
+                }
+                scheduledFuture = getAsyncProfilerExecutor().schedule(
+                        () -> stopWhenSuccess(task), task.getDuration(), TimeUnit.SECONDS
+                );
+            } catch (IOException e) {
+                LOGGER.error("AsyncProfilerTask executor error:" + e.getMessage(), e);
+            }
+        });
+    }
+
+    private void stopWhenError(AsyncProfilerTask task, String errorMessage) {
+        LOGGER.error("AsyncProfilerTask start fail result:" + errorMessage);
```

Comment:
```suggestion
        LOGGER.error("AsyncProfilerTask fails to start: {}"， errorMessage);
```

Correcting the sentence and use log text formater.